experimental/adaptive-dialog/docs/recognizers-rules-steps-reference.md

```diff
@@ -0,0 +1,739 @@
+        }
+    }
+}
+```
+
+## Rules
+_Rules_ enable you to catch and respond to events. The broadest rule is the EventRule that allows you to catch and attach a set of steps to execute when a specific event is emitted by any sub-system. Adaptive dialogs supports couple of other specialized rules to wrap common events that your bot would handle.
+
+Adaptive dialogs support the following Rules - 
+- [Intent rule](#Intent-rule)
+- [Unknown intent rule](#Unknown-intent-rule) 
+- [Event rule](#Event-rule)
+
+At the core, all rules are event handlers. Here are the set of events that can be handled via a rule.
+<a name="events"></a>
+
+| Event               | Description                                       |
+|---------------------|---------------------------------------------------|
+| BeginDialog         | Fired when a dialog is start                      |
+| ActivityReceived    | Fired when a new activity comes in                |
+| RecognizedIntent    | Fired when an intent is recognized                |
+| UnknownIntent       | Fired when an intent is not handled or recognized |
+| StepsStarted        | Fired when a plan is started                      |
+| StepsSaved          | Fires when a plan is saved                        |
+| StepsEnded          | Fires when a plan successful ends                 |
+| StepsResumed        | Fires when a plan is resumed from an interruption |
+| ConsultDialog       | fired when consulting                             |
+| CancelDialog        | fired when dialog canceled                        |
+
+### Intent rule
+Intent rule enables you to catch and react to 'recognizedIntent' event emitted by a recognizer. All built-in recognizers emit this event when they successfully process an input utterance. 
+
+``` C#
+// Create root dialog as an Adaptive dialog.
+var rootDialog = new AdaptiveDialog(nameof(AdaptiveDialog));
+// Add a regex recognizer
+rootDialog.Recognizer = new RegexRecognizer()
+{
+    Intents = new Dictionary<string, string>()
+    {
+        { "HelpIntent", "(?i)help" },
+        { "CancelIntent", "(?i)cancel|never mind" }
+    }
+};
+// Create an intent rule with the intent name
+var helpRule = new IntentRule("HelpIntent");
+
+// Create steps when the bookFlightRule triggers
+var helpSteps = new List<IDialog>();
+helpSteps.Add(new SendActivity("Hello, I'm the samples bot. At the moment, I respond to only help!"));
+helpRule.Steps = helpSteps;
+
+// Add the bookFlight rule to root dialog
+rootDialog.AddRule(helpRule);
+```
+**Note:** You can use the IntentRule to also trigger on 'entities' generated by a recognizer. 
+
+### UnknownIntentRule
+Use this rule to catch and respond to a case when a 'recognizedIntent' event was not caught and handled by any of the other rules. This is especially helpful to capture and handle cases where your dialog wishes to participate in consultation.
```

Comment:
Use of rules in this case is confusing. I've made some updates.

Reply:
after talking to @vishwacsena , I think we need to clarify this a bit. Reading the sample helps to understand. But we need to make sure we explain the relationship between intents and rules in this case